### PR TITLE
Prompt for creds

### PR DESCRIPTION
**Pull request by swivelhinges**
To make it possible to provide credentials for VMs managed by Vagrant from the command line

### CODE REVIEW

File: lib/vagrant-orchestrate/action/setcredentials.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        def apply_creds(username, password)
+          configs = [@machine.config.winrm, @machine.config.ssh]
+          configs.each do |config|
+            next unless config
+            config.username = username
+            config.password = password
+          end
+        end
+
+        def prompt_username
```
Comment (maclennann) [inline lib/vagrant-orchestrate/action/setcredentials.rb]: For Windows boxes (or Linux boxes in Likewise), the username will need the domain prepended. I've been doing it in my own Vagrantfiles with:

```ruby
  best_guess = ENV['USER']
  best_guess = "#{ENV['USERDOMAIN']}\\#{ENV['USERNAME']}" if ENV['USERDOMAIN']
  ask("Please enter your username:"){ |q| q.default = best_guess }
```

**Comment by maclennann** [comment]
Looks pretty alright. Mind adding some usage info to the `README`?

Come to that, the readme is already pretty giant. Is this the time to discuss knowledge management? Use GIthub wiki? Break it up into a docs folder? Keep growing the README (unbounded or until unspecified time)?

**Comment by chrisbaldauf** [comment]
@maclennann, agree that we should have some docs for this feature, as it has some inherent complexity. As far as changing the docs, I'm ok with growing the readme for now. It is long, but it does have a lot of code blocks in it and is reasonably broken up into sections.

File: templates/vagrant/Vagrantfile.erb [hunk 2]
```diff
@@ -14,7 +14,12 @@ Vagrant.configure("2") do |config|
   # This disables up, provision, reload, and destroy for managed servers. Use
   # `vagrant orchestrate push` to communicate with managed servers.
   config.orchestrate.filter_managed_commands = true
-
+  <% if creds_prompt -%>
+  config.orchestrate.credentials do |creds|
```
Comment (chrisbaldauf) [inline templates/vagrant/Vagrantfile.erb]: I think this is worth making a one liner unless there is other stuff that you're planning to put in the bootstrapping.

File: lib/vagrant-orchestrate/command/init.rb [hunk 3]
```diff
@@ -102,6 +102,10 @@ def execute
             o.on("-f", "--force", "Force overwriting of files") do
               options[:force] = true
             end
+
+            o.on("--credentials-prompt", "Prompt for credentials when performing orchestrate operations") do
```
Comment (chrisbaldauf) [inline lib/vagrant-orchestrate/command/init.rb]: Let's add --credentials-file-path as an init parameter as well.

**Pull request merged**